### ISSUE TITLE
Cannot import ONNX models exported from Chainer models with Linear's nobias option

### ISSUE THREAD

**Issue by tterada** [chainer/onnx-chainer#103]
Hello.

I work on verification of DNN inference via ONNX models.
In the progress, I found the following problem.

When exporting a ONNX model from Chainer model with nobias=True of chainer.links.Linear,
Gemm's bias shape is as below.

![2019-02-24 8 25 13](https://user-images.githubusercontent.com/8768366/53292913-e0472500-380d-11e9-8c48-93620c0839c1.png)

In this case, the following import error occurs on Menoh and Intel OpenVINO model optimizer.
`libc++abi.dylib: terminating with uncaught exception of type menoh::error: menoh unsupported operator error: Gemm
Abort trap: 6`

When exporting a ONNX model from Chainer model with nobias=False of chainer.links.Linear,
Gemm's bias shape is as below.

![2019-02-24 8 25 45](https://user-images.githubusercontent.com/8768366/53293045-fc4bc600-380f-11e9-83c4-fc591416e512.png)

In this case, import and inference succeed on Menoh and Intel OpenVINO.

I tried the following modification and confirmed that import and inference run properly.

    def convert_LinearFunction(func, onnx_op_name, opset_version, input_names, output_names, parameters):
        # When the func has bias
        if len(func.inputs) == 2:
            batchsize = func.inputs[0].shape[0]
            bias_dim = func.inputs[1].shape[0]
            if batch size == 1:
                bias = np.zeros((bias_dim), dtype=np.float32)
            else:
                bias = np.zeros((batchsize, bias_dim), dtype=np.float32)
            bias_param = chainer.Parameter(bias)
            parameters.append(bias_param)
            input_names.append(str(id(bias_param)))

**Comment by disktnk** [comment]
Thank you for detail report and helpful patch!

Official definition of `Gemm` operator says

> C: T
> Input tensor C. The shape of C should be unidirectional broadcastable to (M, N).

ref: https://github.com/onnx/onnx/blob/master/docs/Operators.md#gemm

so, shape of input C, both (M,N) and (N,) are accepted to follow the definition.  Actually onnxruntime accepts the both shape. If I misunderstand something, could you please tell me?

**Comment by tterada** [comment]
Thank you for your reply.

Your understanding is probably correct.
I understood the official definition of Gemm operator.

When I tried ONNX-Tensorflow and MXNet to import onnx models exported from chainer models with nobias=True of chainer.links.Linear, import and inference succeeded.
(I used the VGG16 with BatchNormalization models.)

So, it might be a problem with Menoh and Intel OpenVINO.
I don't know what the source code of the corresponding part of Menoh is,
but Intel OpenVINO model optimizer obviously supports only (N,) as the shape of bias.

Perhaps Menoh and Intel OpenVINO doesn't correspond to importing Gemm operator perfectly.

**Comment by shinh** [comment]
I agree we could conclude this is a limitation of importers. That said, now I think it's better not to emit batch dimension because of the reason I wrote in https://github.com/chainer/onnx-chainer/pull/109

What do you think?

**Comment by disktnk** [comment]
As the review comment of PR #109, ONNX-Chainerui do not have to expand dims.

**Comment by disktnk** [comment]
@tterada PR #109 is merged, reported error would be resolved on master branch (or next release), thanks for reporting.

**Comment by tterada** [comment]
OK. 
Thank you for your quick response.
